### ISSUE TITLE
libcxxClang doesn't seem to work as a toolchain

### ISSUE THREAD

**Issue by iphydf** [tweag/rules_nixpkgs#445]
**Describe the bug**
Using clang with libc++ fails to find `<cxxabi.h>`.

**To Reproduce**
Use the following in WORKSPACE:
```
nixpkgs_cc_configure(
    attribute_path = "llvmPackages_16.libcxxClang",
    repository = "@nixpkgs",
)
```
Try to compile something that includes `cxxabi.h` (e.g. googletest) and observe:
```
external/com_google_googletest/googletest/include/gtest/internal/gtest-type-util.h:49:10: fatal error: 'cxxabi.h' file not found
#include <cxxabi.h>
         ^~~~~~~~~~
1 error generated.
```

**Expected behavior**
Compilation should succeed.

**Environment**
 - OS name + version: NixOS 23.05
 - Version of the code: 0.10.0 (loaded by rules_haskell).

**Additional context**
These files definitely exist:
```
/nix/store/6bi80vsz2aa0r5j9y16pcz3gkmx80qfj-libcxxabi-16.0.1-dev/include/cxxabi.h
/nix/store/6bi80vsz2aa0r5j9y16pcz3gkmx80qfj-libcxxabi-16.0.1-dev/include/c++/v1/cxxabi.h
```

**Comment by avdv** [comment]
Is this the same problem as https://github.com/NixOS/nixpkgs/issues/150655 ?

Ie. does passing `--cxxopt=-x --cxxopt=c++` help?

**Comment by iphydf** [comment]
No, all other C++ standard library headers can be found. Also, `-x c++` was already passed by the toolchain. I can see:
```
/nix/store/y02xngjcbhfgliijh2wnpksc3dqhva56-clang-wrapper-16.0.1/bin/cc -U_FORTIFY_SOURCE -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer -x c++ '-std=c++0x' -MD -MF [...]
```

**Comment by avdv** [comment]
I think this is a consequence of nixpkgs having split the toolchain into different packages and libcxxabi not directly being part of the toolchain by default.

A nix-shell provides an environment where these packages are combined again, using the `NIX_CFLAGS_COMPILE` / `NIX_LDFLAGS` variables which are respected by the clang wrapper.

Although this is also not enough, when trying to run `nix-shell -p llvmPackages_16.libcxxClang --run "clang++ -E - <<<'#include <cxxabi.h>' > /dev/null"` it bails out with 
```
In file included from <stdin>:1:
/nix/store/8qwwfr56i3sy5b22a0njp4k3anrqb8c2-libcxxabi-16.0.6-dev/include/cxxabi.h:20:10: fatal error: '__cxxabi_config.h' file not found
#include <__cxxabi_config.h>
         ^~~~~~~~~~~~~~~~~~~
1 error generated.
```

You could make this work by using `wrapCCWith`, like this:
https://github.com/tweag/rules_nixpkgs/blob/2c767691d12b66a92f231bccb06bcf9f7477b962/toolchains/cc/cc.nix#L27-L47

Adding in any library and include directories needed.
 
I am not sure how we could provide a better experience here, we probably would have to run the setupHooks somehow in order to realize the env variables so that the resulting paths would be reflected in the toolchain.
